### ISSUE TITLE
cURL error 7: Failed to connect to www.domain.tld port 443 after 1 ms: Couldn't connect to server

### ISSUE THREAD

**Issue by NetSecond** [akeeba/panopticon#825]
### Due diligence

- [X] I have searched all open and closed issues for this problem.
- [X] I have searched all open and closed discussions for this problem.
- [X] I have looked at the CHANGELOG to make sure this problem has not already been addressed.
- [X] I agree to the Code of Conduct of this repository

### Describe the issue

I have transfered my complete Joomla 5.2 Site to a new Server.
Everything works fine after transfer but if I start Panopticon, I get following Error in the overview:

>> An error occurred while updating the site information.

Connection testers output:

```
GuzzleHttp\Exception\ConnectException

cURL error 7: Failed to connect to www.domain.tld port 443 after 2 ms: Couldn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://www.domain.tld/api/index.php/v1/extensions

/var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:275

#0 /var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(205): GuzzleHttp\Handler\CurlFactory::createRejection()
#1 /var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(157): GuzzleHttp\Handler\CurlFactory::finishError()
#2 /var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/CurlHandler.php(47): GuzzleHttp\Handler\CurlFactory::finish()
#3 /var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php(28): GuzzleHttp\Handler\CurlHandler->__invoke()
#4 /var/www/pan/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php(48): GuzzleHttp\Handler\Proxy::GuzzleHttp\Handler\{closure}()
#5 /var/www/pan/vendor/guzzlehttp/guzzle/src/PrepareBodyMiddleware.php(35): GuzzleHttp\Handler\Proxy::GuzzleHttp\Handler\{closure}()
#6 /var/www/pan/vendor/guzzlehttp/guzzle/src/Middleware.php(31): GuzzleHttp\PrepareBodyMiddleware->__invoke()
#7 /var/www/pan/vendor/guzzlehttp/guzzle/src/RedirectMiddleware.php(71): GuzzleHttp\Middleware::GuzzleHttp\{closure}()
#8 /var/www/pan/vendor/guzzlehttp/guzzle/src/Middleware.php(63): GuzzleHttp\RedirectMiddleware->__invoke()
#9 /var/www/pan/vendor/guzzlehttp/guzzle/src/HandlerStack.php(75): GuzzleHttp\Middleware::GuzzleHttp\{closure}()
#10 /var/www/pan/vendor/guzzlehttp/guzzle/src/Client.php(333): GuzzleHttp\HandlerStack->__invoke()
#11 /var/www/pan/vendor/guzzlehttp/guzzle/src/Client.php(169): GuzzleHttp\Client->transfer()
#12 /var/www/pan/vendor/guzzlehttp/guzzle/src/Client.php(189): GuzzleHttp\Client->requestAsync()
#13 /var/www/pan/vendor/guzzlehttp/guzzle/src/ClientTrait.php(44): GuzzleHttp\Client->request()
#14 /var/www/pan/src/Model/Trait/SiteTestConnectionJoomlaTrait.php(69): GuzzleHttp\Client->get()
#15 /var/www/pan/src/Model/Site.php(354): Akeeba\Panopticon\Model\Site->testConnectionJoomla()
#16 /var/www/pan/src/Controller/Sites.php(131): Akeeba\Panopticon\Model\Site->testConnection()
#17 /var/www/pan/vendor/akeeba/awf/src/Mvc/Controller.php(375): Akeeba\Panopticon\Controller\Sites->connectionDoctor()
#18 /var/www/pan/vendor/akeeba/awf/src/Mvc/DataController.php(57): Awf\Mvc\Controller->execute()
#19 /var/www/pan/src/Controller/Sites.php(64): Awf\Mvc\DataController->execute()
#20 /var/www/pan/vendor/akeeba/awf/src/Dispatcher/Dispatcher.php(125): Akeeba\Panopticon\Controller\Sites->execute()
#21 /var/www/pan/src/Dispatcher.php(36): Awf\Dispatcher\Dispatcher->dispatch()
#22 /var/www/pan/vendor/akeeba/awf/src/Application/Application.php(247): Akeeba\Panopticon\Dispatcher->dispatch()
#23 /var/www/pan/src/Application.php(430): Awf\Application\Application->dispatch()
#24 /var/www/pan/index.php(35): Akeeba\Panopticon\Application->dispatch()
#25 /var/www/pan/index.php(38): {closure}()
#26 {main}
```

Any ideas ??

Since this error only occurs with this one domain, I assume that Panopticon itself is running correctly.

Panopticon Connector v.1.0.3
API Level 101

### Panopticon version

1.2.2

### PHP version

8.1.3

### Database type

MariaDB

### Database version

??

### Browser

None

**Comment by NetSecond** [comment]
Panopticon Connector 1.0.6 installed, no change ....

**Comment by nikosdion** [comment]
As per the error message, cURL cannot connect to your remote site. This has nothing to do with the code in the product and everything to do with either your server environment. Maybe this will help you understand this https://stackoverflow.com/questions/9922562/how-to-resolve-curl-error-7-couldnt-connect-to-host

**Comment by NetSecond** [comment]
Ok, I know, I wrote

>> ....  I assume that Panopticon itself is running correctly.

I hoped, that you can give me a tip, where I have to search ....
So I will try to fix it with your url to stackoverflow.

Thanks and regards

**Comment by NetSecond** [comment]
I found the problem:

The LetsEncrypt paths in the configuration for Postfix and Dovecot were no longer correct.

I combined several certificates from a domain with its subdomains into one, which worked without any problems.
However, I still had the old path in Postfix and Dovecot and noticed in the mail client that the server could not retrieve a mailbox.
I then adjusted the paths in Postfix and Dovecot and suddenly curl could access the API again.
The rest (Apache & Co.) was already running beforehand.

Crazy ....

**Comment by nikosdion** [comment]
That doesn't make much sense indeed. Postfix handles SMTP and Dovecot handles IMAP. They should have no bearing on the HTTP services of your server. I think that whatever you did had an unintended effect which solved your issue. The two most plausible scenarios I can think of is restarting the service which automatically updates the TLS certificates (in most cases it'd be `certbot` which updates certificates with Let's Encrypt), or restarting the web server service which now loaded an updated TLS certificate (after updating the TLS certificate you need to reload or restart the web server system service for it to “see” the certificates). Again, these are the most plausible but not the only possibly ones.

**Comment by NetSecond** [comment]
I know, everything went perfectly and everything was up to date.
All services were restarted, including the entire server for testing purposes.
Only when I noticed that the SSL/TLS path in the Postfix and Dovecot configuration was wrong and I corrected it did curl finally run again.
I then neither generated new certificates nor changed anything else, but only adjusted the two mail server certificate paths and reloaded Postfix and Dovecot.
The websites were already running and certbot adjusted the vhosts and everything was ok, except for mail and curl.
Well, only after correcting the path of Postfix and Dovecot did curl work again.
That doesn't make sense, but it was reproducible.
